src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -1,55 +1,29 @@
 package seedu.address.logic.commands;
 
-import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.FLAG_EVENT;
+import static seedu.address.logic.parser.CliSyntax.FLAG_PERSON;
+import static seedu.address.logic.parser.CliSyntax.FLAG_TRAINING;
 
-import java.util.List;
-
-import seedu.address.commons.core.Messages;
-import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
-import seedu.address.model.person.Person;
 
 /**
- * Deletes a person identified using it's displayed index from the address book.
+ * Deletes a person, event, or training.
  */
-public class DeleteCommand extends Command {
+public abstract class DeleteCommand extends Command {
```

Comment:
Wow! 😯Great idea  💡making this an abstract class! I  🙋‍♂️ personally wouldn't have thought 💭 of it myself! This will make implementing the delete training command a lot easier!